Video game analytics
===
ToDo: Explain the concept, source

In [5]:
from pandas import read_csv, to_numeric

file = read_csv('Video_Games_Sales_as_at_22_Dec_2016.csv')
file.head(10)

<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Platform</th>
      <th>Year_of_Release</th>
      <th>Genre</th>
      <th>Publisher</th>
      <th>NA_Sales</th>
      <th>EU_Sales</th>
      <th>JP_Sales</th>
      <th>Other_Sales</th>
      <th>Global_Sales</th>
      <th>Critic_Score</th>
      <th>Critic_Count</th>
      <th>User_Score</th>
      <th>User_Count</th>
      <th>Developer</th>
      <th>Rating</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Wii Sports</td>
      <td>Wii</td>
      <td>2006.0</td>
      <td>Sports</td>
      <td>Nintendo</td>
      <td>41.36</td>
      <td>28.96</td>
      <td>3.77</td>
      <td>8.45</td>
      <td>82.53</td>
      <td>76.0</td>
      <td>51.0</td>
      <td>8</td>
      <td>322.0</td>
      <td>Nintendo</td>
      <td>E</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Super Mario Bros.</td>
      <td>NES</td>
      <td>1985.0</td>
      <td>Platform</td>
      <td>Nintendo</td>
      <td>29.08</td>
      <td>3.58</td>
      <td>6.81</td>
      <td>0.77</td>
      <td>40.24</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Mario Kart Wii</td>
      <td>Wii</td>
      <td>2008.0</td>
      <td>Racing</td>
      <td>Nintendo</td>
      <td>15.68</td>
      <td>12.76</td>
      <td>3.79</td>
      <td>3.29</td>
      <td>35.52</td>
      <td>82.0</td>
      <td>73.0</td>
      <td>8.3</td>
      <td>709.0</td>
      <td>Nintendo</td>
      <td>E</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Wii Sports Resort</td>
      <td>Wii</td>
      <td>2009.0</td>
      <td>Sports</td>
      <td>Nintendo</td>
      <td>15.61</td>
      <td>10.93</td>
      <td>3.28</td>
      <td>2.95</td>
      <td>32.77</td>
      <td>80.0</td>
      <td>73.0</td>
      <td>8</td>
      <td>192.0</td>
      <td>Nintendo</td>
      <td>E</td>
    </tr>
    <tr>
      <th>4</th>
      <td>Pokemon Red/Pokemon Blue</td>
      <td>GB</td>
      <td>1996.0</td>
      <td>Role-Playing</td>
      <td>Nintendo</td>
      <td>11.27</td>
      <td>8.89</td>
      <td>10.22</td>
      <td>1.00</td>
      <td>31.37</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>5</th>
      <td>Tetris</td>
      <td>GB</td>
      <td>1989.0</td>
      <td>Puzzle</td>
      <td>Nintendo</td>
      <td>23.20</td>
      <td>2.26</td>
      <td>4.22</td>
      <td>0.58</td>
      <td>30.26</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>6</th>
      <td>New Super Mario Bros.</td>
      <td>DS</td>
      <td>2006.0</td>
      <td>Platform</td>
      <td>Nintendo</td>
      <td>11.28</td>
      <td>9.14</td>
      <td>6.50</td>
      <td>2.88</td>
      <td>29.80</td>
      <td>89.0</td>
      <td>65.0</td>
      <td>8.5</td>
      <td>431.0</td>
      <td>Nintendo</td>
      <td>E</td>
    </tr>
    <tr>
      <th>7</th>
      <td>Wii Play</td>
      <td>Wii</td>
      <td>2006.0</td>
      <td>Misc</td>
      <td>Nintendo</td>
      <td>13.96</td>
      <td>9.18</td>
      <td>2.93</td>
      <td>2.84</td>
      <td>28.92</td>
      <td>58.0</td>
      <td>41.0</td>
      <td>6.6</td>
      <td>129.0</td>
      <td>Nintendo</td>
      <td>E</td>
    </tr>
    <tr>
      <th>8</th>
      <td>New Super Mario Bros. Wii</td>
      <td>Wii</td>
      <td>2009.0</td>
      <td>Platform</td>
      <td>Nintendo</td>
      <td>14.44</td>
      <td>6.94</td>
      <td>4.70</td>
      <td>2.24</td>
      <td>28.32</td>
      <td>87.0</td>
      <td>80.0</td>
      <td>8.4</td>
      <td>594.0</td>
      <td>Nintendo</td>
      <td>E</td>
    </tr>
    <tr>
      <th>9</th>
      <td>Duck Hunt</td>
      <td>NES</td>
     

ToDo: Explain data and pandas

In [13]:
file.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16719 entries, 0 to 16718
Data columns (total 16 columns):
Name               16717 non-null object
Platform           16719 non-null object
Year_of_Release    16450 non-null float64
Genre              16717 non-null object
Publisher          16665 non-null object
NA_Sales           16719 non-null float64
EU_Sales           16719 non-null float64
JP_Sales           16719 non-null float64
Other_Sales        16719 non-null float64
Global_Sales       16719 non-null float64
Critic_Score       8137 non-null float64
Critic_Count       8137 non-null float64
User_Score         10015 non-null object
User_Count         7590 non-null float64
Developer          10096 non-null object
Rating             9950 non-null object
dtypes: float64(9), object(7)
memory usage: 2.0+ MB


ToDo: User Score not detected as float

In [7]:
file.describe()

<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Year_of_Release</th>
      <th>NA_Sales</th>
      <th>EU_Sales</th>
      <th>JP_Sales</th>
      <th>Other_Sales</th>
      <th>Global_Sales</th>
      <th>Critic_Score</th>
      <th>Critic_Count</th>
      <th>User_Count</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>count</th>
      <td>16450.000000</td>
      <td>16719.000000</td>
      <td>16719.000000</td>
      <td>16719.000000</td>
      <td>16719.000000</td>
      <td>16719.000000</td>
      <td>8137.000000</td>
      <td>8137.000000</td>
      <td>7590.000000</td>
    </tr>
    <tr>
      <th>mean</th>
      <td>2006.487356</td>
      <td>0.263330</td>
      <td>0.145025</td>
      <td>0.077602</td>
      <td>0.047332</td>
      <td>0.533543</td>
      <td>68.967679</td>
      <td>26.360821</td>
      <td>162.229908</td>
    </tr>
    <tr>
      <th>std</th>
      <td>5.878995</td>
      <td>0.813514</td>
      <td>0.503283</td>
      <td>0.308818</td>
      <td>0.186710</td>
      <td>1.547935</td>
      <td>13.938165</td>
      <td>18.980495</td>
      <td>561.282326</td>
    </tr>
    <tr>
      <th>min</th>
      <td>1980.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.010000</td>
      <td>13.000000</td>
      <td>3.000000</td>
      <td>4.000000</td>
    </tr>
    <tr>
      <th>25%</th>
      <td>2003.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.060000</td>
      <td>60.000000</td>
      <td>12.000000</td>
      <td>10.000000</td>
    </tr>
    <tr>
      <th>50%</th>
      <td>2007.000000</td>
      <td>0.080000</td>
      <td>0.020000</td>
      <td>0.000000</td>
      <td>0.010000</td>
      <td>0.170000</td>
      <td>71.000000</td>
      <td>21.000000</td>
      <td>24.000000</td>
    </tr>
    <tr>
      <th>75%</th>
      <td>2010.000000</td>
      <td>0.240000</td>
      <td>0.110000</td>
      <td>0.040000</td>
      <td>0.030000</td>
      <td>0.470000</td>
      <td>79.000000</td>
      <td>36.000000</td>
      <td>81.000000</td>
    </tr>
    <tr>
      <th>max</th>
      <td>2020.000000</td>
      <td>41.360000</td>
      <td>28.960000</td>
      <td>10.220000</td>
      <td>10.570000</td>
      <td>82.530000</td>
      <td>98.000000</td>
      <td>113.000000</td>
      <td>10665.000000</td>
    </tr>
  </tbody>
</table>
</div>

ToDo 

In [8]:
file[file['Year_of_Release'] > 2017]

<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Platform</th>
      <th>Year_of_Release</th>
      <th>Genre</th>
      <th>Publisher</th>
      <th>NA_Sales</th>
      <th>EU_Sales</th>
      <th>JP_Sales</th>
      <th>Other_Sales</th>
      <th>Global_Sales</th>
      <th>Critic_Score</th>
      <th>Critic_Count</th>
      <th>User_Score</th>
      <th>User_Count</th>
      <th>Developer</th>
      <th>Rating</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>5936</th>
      <td>Imagine: Makeup Artist</td>
      <td>DS</td>
      <td>2020.0</td>
      <td>Simulation</td>
      <td>Ubisoft</td>
      <td>0.27</td>
      <td>0.0</td>
      <td>0.0</td>
      <td>0.02</td>
      <td>0.29</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>tbd</td>
      <td>NaN</td>
      <td>Ubisoft</td>
      <td>E</td>
    </tr>
  </tbody>
</table>
</div>

ToDo: Some games have no scores

In [9]:
len(file[file['Global_Sales'] != file['NA_Sales'] + file['EU_Sales'] + file['JP_Sales'] + file['Other_Sales']])

6814

ToDo: Global sales rounding issues (not significant)

In [9]:
from pandas import to_numeric

file['User_Score'].apply(to_numeric)

ValueError: Unable to parse string "tbd" at position 0

ToDo: User_Score not converted to float because "tbd"

In [ ]:
games_without_ratings = file[file['User_Score'] == "tbd"]
games_without_ratings[['Name', 'Platform', 'Year_of_Release', 'Global_Sales', 'Critic_Score', 'User_Score', 'User_Count']].head(15)

<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Platform</th>
      <th>Year_of_Release</th>
      <th>Global_Sales</th>
      <th>Critic_Score</th>
      <th>User_Score</th>
      <th>User_Count</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>119</th>
      <td>Zumba Fitness</td>
      <td>Wii</td>
      <td>2010.0</td>
      <td>6.71</td>
      <td>NaN</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>301</th>
      <td>Namco Museum: 50th Anniversary</td>
      <td>PS2</td>
      <td>2005.0</td>
      <td>3.98</td>
      <td>61.0</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>520</th>
      <td>Zumba Fitness 2</td>
      <td>Wii</td>
      <td>2011.0</td>
      <td>2.81</td>
      <td>NaN</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>645</th>
      <td>uDraw Studio</td>
      <td>Wii</td>
      <td>2010.0</td>
      <td>2.42</td>
      <td>71.0</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>657</th>
      <td>Frogger's Adventures: Temple of the Frog</td>
      <td>GBA</td>
      <td>NaN</td>
      <td>2.39</td>
      <td>73.0</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>718</th>
      <td>Just Dance Kids</td>
      <td>Wii</td>
      <td>2010.0</td>
      <td>2.24</td>
      <td>NaN</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>726</th>
      <td>Dance Dance Revolution X2</td>
      <td>PS2</td>
      <td>2009.0</td>
      <td>2.23</td>
      <td>NaN</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>821</th>
      <td>The Incredibles</td>
      <td>GBA</td>
      <td>2004.0</td>
      <td>2.06</td>
      <td>55.0</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>881</th>
      <td>Who wants to be a millionaire</td>
      <td>PC</td>
      <td>1999.0</td>
      <td>1.94</td>
      <td>NaN</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1047</th>
      <td>Tetris Worlds</td>
      <td>GBA</td>
      <td>2001.0</td>
      <td>1.71</td>
      <td>65.0</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1048</th>
      <td>Imagine: Teacher</td>
      <td>DS</td>
      <td>2008.0</td>
      <td>1.71</td>
      <td>NaN</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1069</th>
      <td>Personal Trainer: Math</td>
      <td>DS</td>
      <td>2007.0</td>
      <td>1.69</td>
      <td>63.0</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1070</th>
      <td>Game Party 3</td>
      <td>Wii</td>
      <td>2009.0</td>
      <td>1.68</td>
      <td>37.0</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1089</th>
      <td>Monsters, Inc.</td>
      <td>GBA</td>
      <td>2001.0</td>
      <td>1.67</td>
      <td>NaN</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>1151</th>
      <td>SpongeBob's Atlantis SquarePantis</td>
      <td>DS</td>
      <td>2007.0</td>
      <td>1.60</td>
      <td>64.0</td>
      <td>tbd</td>
      <td>NaN</td>
    </tr>
  </tbody>
</table>
</div>

ToDo: Some games have no scores

In [14]:
games_with_ratings = file[file['User_Count'] > 0][file['Critic_Count'] > 0]
len(games_with_ratings)

F:\Anaconda3\lib\site-packages\ipykernel\__main__.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  if __name__ == '__main__':


7017

ToDo: Describe games_with_ratings

In [15]:
games_with_ratings['User_Score'] = games_with_ratings['User_Score'].apply(to_numeric)
games_with_ratings['User_Score'] = games_with_ratings['User_Score'] * 10
games_with_ratings.describe()

<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Year_of_Release</th>
      <th>NA_Sales</th>
      <th>EU_Sales</th>
      <th>JP_Sales</th>
      <th>Other_Sales</th>
      <th>Global_Sales</th>
      <th>Critic_Score</th>
      <th>Critic_Count</th>
      <th>User_Score</th>
      <th>User_Count</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>count</th>
      <td>6894.000000</td>
      <td>7017.000000</td>
      <td>7017.000000</td>
      <td>7017.000000</td>
      <td>7017.000000</td>
      <td>7017.000000</td>
      <td>7017.000000</td>
      <td>7017.000000</td>
      <td>7017.000000</td>
      <td>7017.000000</td>
    </tr>
    <tr>
      <th>mean</th>
      <td>2007.482303</td>
      <td>0.389290</td>
      <td>0.233095</td>
      <td>0.062951</td>
      <td>0.081525</td>
      <td>0.767049</td>
      <td>70.249822</td>
      <td>28.783241</td>
      <td>71.824284</td>
      <td>173.432664</td>
    </tr>
    <tr>
      <th>std</th>
      <td>4.236401</td>
      <td>0.957051</td>
      <td>0.679210</td>
      <td>0.284162</td>
      <td>0.266594</td>
      <td>1.940317</td>
      <td>13.880646</td>
      <td>19.167119</td>
      <td>14.412410</td>
      <td>581.977516</td>
    </tr>
    <tr>
      <th>min</th>
      <td>1985.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.000000</td>
      <td>0.010000</td>
      <td>13.000000</td>
      <td>3.000000</td>
      <td>5.000000</td>
      <td>4.000000</td>
    </tr>
    <tr>
      <th>25%</th>
      <td>2004.000000</td>
      <td>0.060000</td>
      <td>0.020000</td>
      <td>0.000000</td>
      <td>0.010000</td>
      <td>0.110000</td>
      <td>62.000000</td>
      <td>14.000000</td>
      <td>65.000000</td>
      <td>11.000000</td>
    </tr>
    <tr>
      <th>50%</th>
      <td>2007.000000</td>
      <td>0.150000</td>
      <td>0.060000</td>
      <td>0.000000</td>
      <td>0.020000</td>
      <td>0.290000</td>
      <td>72.000000</td>
      <td>24.000000</td>
      <td>75.000000</td>
      <td>27.000000</td>
    </tr>
    <tr>
      <th>75%</th>
      <td>2011.000000</td>
      <td>0.390000</td>
      <td>0.210000</td>
      <td>0.010000</td>
      <td>0.070000</td>
      <td>0.750000</td>
      <td>80.000000</td>
      <td>39.000000</td>
      <td>82.000000</td>
      <td>89.000000</td>
    </tr>
    <tr>
      <th>max</th>
      <td>2016.000000</td>
      <td>41.360000</td>
      <td>28.960000</td>
      <td>6.500000</td>
      <td>10.570000</td>
      <td>82.530000</td>
      <td>98.000000</td>
      <td>113.000000</td>
      <td>96.000000</td>
      <td>10665.000000</td>
    </tr>
  </tbody>
</table>
</div>

## How are critic scores different from user scores? Which game has the highest delta?

In [31]:
games_with_ratings["deltaScore"] = games_with_ratings["Critic_Score"] - games_with_ratings["User_Score"]
games_with_ratings.head(15)

<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Platform</th>
      <th>Year_of_Release</th>
      <th>Genre</th>
      <th>Publisher</th>
      <th>NA_Sales</th>
      <th>EU_Sales</th>
      <th>JP_Sales</th>
      <th>Other_Sales</th>
      <th>Global_Sales</th>
      <th>Critic_Score</th>
      <th>Critic_Count</th>
      <th>User_Score</th>
      <th>User_Count</th>
      <th>Developer</th>
      <th>Rating</th>
      <th>delta score</th>
      <th>deltaScore</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Wii Sports</td>
      <td>Wii</td>
      <td>2006.0</td>
      <td>Sports</td>
      <td>Nintendo</td>
      <td>41.36</td>
      <td>28.96</td>
      <td>3.77</td>
      <td>8.45</td>
      <td>82.53</td>
      <td>76.0</td>
      <td>51.0</td>
      <td>80.0</td>
      <td>322.0</td>
      <td>Nintendo</td>
      <td>E</td>
      <td>-4.0</td>
      <td>-4.0</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Mario Kart Wii</td>
      <td>Wii</td>
      <td>2008.0</td>
      <td>Racing</td>
      <td>Nintendo</td>
      <td>15.68</td>
      <td>12.76</td>
      <td>3.79</td>
      <td>3.29</td>
      <td>35.52</td>
      <td>82.0</td>
      <td>73.0</td>
      <td>83.0</td>
      <td>709.0</td>
      <td>Nintendo</td>
      <td>E</td>
      <td>-1.0</td>
      <td>-1.0</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Wii Sports Resort</td>
      <td>Wii</td>
      <td>2009.0</td>
      <td>Sports</td>
      <td>Nintendo</td>
      <td>15.61</td>
      <td>10.93</td>
      <td>3.28</td>
      <td>2.95</td>
      <td>32.77</td>
      <td>80.0</td>
      <td>73.0</td>
      <td>80.0</td>
      <td>192.0</td>
      <td>Nintendo</td>
      <td>E</td>
      <td>0.0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>6</th>
      <td>New Super Mario Bros.</td>
      <td>DS</td>
      <td>2006.0</td>
      <td>Platform</td>
      <td>Nintendo</td>
      <td>11.28</td>
      <td>9.14</td>
      <td>6.50</td>
      <td>2.88</td>
      <td>29.80</td>
      <td>89.0</td>
      <td>65.0</td>
      <td>85.0</td>
      <td>431.0</td>
      <td>Nintendo</td>
      <td>E</td>
      <td>4.0</td>
      <td>4.0</td>
    </tr>
    <tr>
      <th>7</th>
      <td>Wii Play</td>
      <td>Wii</td>
      <td>2006.0</td>
      <td>Misc</td>
      <td>Nintendo</td>
      <td>13.96</td>
      <td>9.18</td>
      <td>2.93</td>
      <td>2.84</td>
      <td>28.92</td>
      <td>58.0</td>
      <td>41.0</td>
      <td>66.0</td>
      <td>129.0</td>
      <td>Nintendo</td>
      <td>E</td>
      <td>-8.0</td>
      <td>-8.0</td>
    </tr>
    <tr>
      <th>8</th>
      <td>New Super Mario Bros. Wii</td>
      <td>Wii</td>
      <td>2009.0</td>
      <td>Platform</td>
      <td>Nintendo</td>
      <td>14.44</td>
      <td>6.94</td>
      <td>4.70</td>
      <td>2.24</td>
      <td>28.32</td>
      <td>87.0</td>
      <td>80.0</td>
      <td>84.0</td>
      <td>594.0</td>
      <td>Nintendo</td>
      <td>E</td>
      <td>3.0</td>
      <td>3.0</td>
    </tr>
    <tr>
      <th>11</th>
      <td>Mario Kart DS</td>
      <td>DS</td>
      <td>2005.0</td>
      <td>Racing</td>
      <td>Nintendo</td>
      <td>9.71</td>
      <td>7.47</td>
      <td>4.13</td>
      <td>1.90</td>
      <td>23.21</td>
      <td>91.0</td>
      <td>64.0</td>
      <td>86.0</td>
      <td>464.0</td>
      <td>Nintendo</td>
      <td>E</td>
      <td>5.0</td>
      <td>5.0</td>
    </tr>
    <tr>
      <th>13</th>
      <td>Wii Fit</td>
      <td>Wii</td>
      <td>2007.0</td>
      <td>Sports</td>
      <td>Nintendo</td>
      <td>8.92</td>
      <td>8.03</td>
      <td>3.60</td>
      <td>2.15</td>
      <td>22.70</td>
      <td>80.0</td>
      <td>63.0</td>
      <td>77.0</td>
      <td>146.0</td>
      <td>Nintendo</td>
      <td>E</td>
      <td>3.0</td>
      <td>3.0</td>
    </tr>
    <tr>
      <th>14</th>
      <td>Kinect Adventures!</td>
      <td>X360<

In [46]:
games_with_ratings["deltaScore"].describe()

count    7017.000000
mean       -1.574462
std        12.960576
min       -62.000000
25%        -9.000000
50%        -2.000000
75%         5.000000
max        66.000000
Name: deltaScore, dtype: float64

In [41]:
len(games_with_ratings[games_with_ratings["deltaScore"] < 0])

3863

In [40]:
len(games_with_ratings[games_with_ratings["deltaScore"] > 0])

2859

In [42]:
len(games_with_ratings[games_with_ratings["deltaScore"] == 0])

295

Of the 7017 games that have a score, **3863** games have a higher user score, **2859** games have a higher critics score. In **295** games the critics and user score are equal.

## Does a publisher have a significant difference in sales of different regions?


In [45]:
file.head(15)

<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Name</th>
      <th>Platform</th>
      <th>Year_of_Release</th>
      <th>Genre</th>
      <th>Publisher</th>
      <th>NA_Sales</th>
      <th>EU_Sales</th>
      <th>JP_Sales</th>
      <th>Other_Sales</th>
      <th>Global_Sales</th>
      <th>Critic_Score</th>
      <th>Critic_Count</th>
      <th>User_Score</th>
      <th>User_Count</th>
      <th>Developer</th>
      <th>Rating</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>Wii Sports</td>
      <td>Wii</td>
      <td>2006.0</td>
      <td>Sports</td>
      <td>Nintendo</td>
      <td>41.36</td>
      <td>28.96</td>
      <td>3.77</td>
      <td>8.45</td>
      <td>82.53</td>
      <td>76.0</td>
      <td>51.0</td>
      <td>8</td>
      <td>322.0</td>
      <td>Nintendo</td>
      <td>E</td>
    </tr>
    <tr>
      <th>1</th>
      <td>Super Mario Bros.</td>
      <td>NES</td>
      <td>1985.0</td>
      <td>Platform</td>
      <td>Nintendo</td>
      <td>29.08</td>
      <td>3.58</td>
      <td>6.81</td>
      <td>0.77</td>
      <td>40.24</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>2</th>
      <td>Mario Kart Wii</td>
      <td>Wii</td>
      <td>2008.0</td>
      <td>Racing</td>
      <td>Nintendo</td>
      <td>15.68</td>
      <td>12.76</td>
      <td>3.79</td>
      <td>3.29</td>
      <td>35.52</td>
      <td>82.0</td>
      <td>73.0</td>
      <td>8.3</td>
      <td>709.0</td>
      <td>Nintendo</td>
      <td>E</td>
    </tr>
    <tr>
      <th>3</th>
      <td>Wii Sports Resort</td>
      <td>Wii</td>
      <td>2009.0</td>
      <td>Sports</td>
      <td>Nintendo</td>
      <td>15.61</td>
      <td>10.93</td>
      <td>3.28</td>
      <td>2.95</td>
      <td>32.77</td>
      <td>80.0</td>
      <td>73.0</td>
      <td>8</td>
      <td>192.0</td>
      <td>Nintendo</td>
      <td>E</td>
    </tr>
    <tr>
      <th>4</th>
      <td>Pokemon Red/Pokemon Blue</td>
      <td>GB</td>
      <td>1996.0</td>
      <td>Role-Playing</td>
      <td>Nintendo</td>
      <td>11.27</td>
      <td>8.89</td>
      <td>10.22</td>
      <td>1.00</td>
      <td>31.37</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>5</th>
      <td>Tetris</td>
      <td>GB</td>
      <td>1989.0</td>
      <td>Puzzle</td>
      <td>Nintendo</td>
      <td>23.20</td>
      <td>2.26</td>
      <td>4.22</td>
      <td>0.58</td>
      <td>30.26</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
      <td>NaN</td>
    </tr>
    <tr>
      <th>6</th>
      <td>New Super Mario Bros.</td>
      <td>DS</td>
      <td>2006.0</td>
      <td>Platform</td>
      <td>Nintendo</td>
      <td>11.28</td>
      <td>9.14</td>
      <td>6.50</td>
      <td>2.88</td>
      <td>29.80</td>
      <td>89.0</td>
      <td>65.0</td>
      <td>8.5</td>
      <td>431.0</td>
      <td>Nintendo</td>
      <td>E</td>
    </tr>
    <tr>
      <th>7</th>
      <td>Wii Play</td>
      <td>Wii</td>
      <td>2006.0</td>
      <td>Misc</td>
      <td>Nintendo</td>
      <td>13.96</td>
      <td>9.18</td>
      <td>2.93</td>
      <td>2.84</td>
      <td>28.92</td>
      <td>58.0</td>
      <td>41.0</td>
      <td>6.6</td>
      <td>129.0</td>
      <td>Nintendo</td>
      <td>E</td>
    </tr>
    <tr>
      <th>8</th>
      <td>New Super Mario Bros. Wii</td>
      <td>Wii</td>
      <td>2009.0</td>
      <td>Platform</td>
      <td>Nintendo</td>
      <td>14.44</td>
      <td>6.94</td>
      <td>4.70</td>
      <td>2.24</td>
      <td>28.32</td>
      <td>87.0</td>
      <td>80.0</td>
      <td>8.4</td>
      <td>594.0</td>
      <td>Nintendo</td>
      <td>E</td>
    </tr>
    <tr>
      <th>9</th>
      <td>Duck Hunt</td>
      <td>NES</td>
     